wordpress-mobile/react-native-libraries-publisher
# Bump React Native `0.73.3` and dependencies

Updates the React Native version to `0.73.3`.

It also updates to node `20`, React Native currently requires at [least version 18](https://reactnative.dev/blog/2023/12/06/0.73-debugging-improvements-stable-symlinks#other-breaking-changes). We use version `20` to match the version we use in Gutenberg.

Since we are updating to Node 20, the `package-lock.json` structure was updated.

Due to some packages having different versions of React, it adds the `legacy-peer-deps` flag to be set when running `npm install`.

Removes exporting the `CLIENT_SIDE_BUILD` variable since this is no longer supported in React Native Reanimated 3.

It also includes the following updates:
- Gradle is updated to `7.5`
- It sets the `kotlinVersion` to `1.8.0`
- `com.automattic.android.publish-to-s3` updated to `0.8.0`

## Testing instructions

Similar to other [previous updates](https://github.com/wordpress-mobile/react-native-libraries-publisher/pull/23):

1. Run the command `npm install`.
2. Run the command `./gradlew publishToMavenLocal -exclude-task prepareToPublishToS3`.
3. Observe that all libraries are built and published successfully in the local folder `~/.m2/repository/org/wordpress/react-native-libraries`.
